packages/core/src/fhir-deduplication/shared.ts

```diff
@@ -400,3 +405,51 @@ export function ensureValidPeriod(period: Period | undefined): Period | undefine
   }
   return period;
 }
+
+export function fetchCodingCodeOrDisplayOrSystem(
+  coding: Coding,
+  field: "code" | "display" | "system"
+): string | undefined {
+  const { log } = out(`fetchCodingCodeOrDisplayOrSystem - coding ${coding}`);
+  try {
+    return coding[field]?.trim().toLowerCase();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  } catch (error: any) {
+    if (error instanceof TypeError) {
+      const msg = `Error fetching field from coding. ${errorToString(error)}`;
```

Comment:
we don't normally attach dynamic properties inside the msg. This will cause a new alert on the Sentry for each different scenario

Reply:
Good call - done.